### ISSUE TITLE
`invalid-name` false positives on generics

### ISSUE THREAD

**Issue by DetachHead** [PyCQA/pylint#5585]
### Bug description

```py
T = TypeVar("T") # Class name "T" doesn't conform to PascalCase naming style ('[^\\W\\da-z][^\\W_]+$' pattern)
```

### Command used

```shell
pylint foo.py
```

### Pylint output

```shell
************* Module foo
foo.py:6:0: C0103: Class name "T" doesn't conform to PascalCase naming style ('[^\\W\\da-z][^\\W_]+$' pattern) (invalid-name)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)]
```

**Comment by DanielNoord** [comment]
Duplicate of #3401. A PR has been opened (#5221) but it's waiting on review. I do think this should be fixed in `2.13` 😄
